### DISCUSSION TITLE
How to get nginx to POST to an endpoint with preconfigured auth and query parameters

### DISCUSSION THREAD

**Discussion by chrislawso** [nginx/nginx#415] [General]
I have an endpoint on lan ip that requires auth and a specific query parameter that I use to trigger actions:
https://username:password@ip.ip.ip.ip:80/api/control?do=start&Location=one&Key=12345

How can I configure nginx so that when someone performs a page load of ie 
https://nginx-fqdn-url.com/test 
then nginx would perform a POST or curl of a preconfigured url
https://username:password@ip.ip.ip.ip:80/api/control?do=start&Location=one&Key=12345

Thank you

**Comment by HanadaLee** [comment]
you can try to use auth_request
https://nginx.org/en/docs/http/ngx_http_auth_request_module.html

**Reply by chrislawso** [reply]
Hi, I do not know how to correctly use auth_request within the configuration I created.

I simplified the config to troubleshoot and made some progress.

Using the below config nginx successfully injects/appends the api query parameters to the end of the url.

```
server {

listen 3000;

# Location block for /api/control

location /api/control {

rewrite ^/api/control$ /api/control?do=key&command=activate break;

# Proxy the request to the backend server

proxy_pass http://service-at-local-ip-address/;

# Set the Authorization header securely

proxy_pass_header Authorization;

# Additional headers for the proxy

proxy_set_header Host $host;

proxy_set_header X-Real-IP $remote_addr;

proxy_set_header X-Forwarded-Proto $scheme;

}

}
```

The above config allows the client to successfully load http://nginx-address/api/control

and nginx successfully forwards the request to http://service-at-local-ip-address/api/control/api/control?do=key&command=activate

In this testing config I am using location /api/control to match the same paths expected by the service-at-local-ip-address and in testing I am unable to find how to change the location to an arbitrary different name ie /testapi

When I change the config to, for example, '''/testapi''' it does not work and returns error "400 Bad Request Your browser sent a request that this server could not understand" when I use the following changes:
```
location /testapi {

rewrite ^/testapi$ /api/control?do=key&command=activate break;
```

The nginx logs identifies:

'''

2025/01/03 02:52:15 [debug] 548#548: *26 http output filter "/testapi?"

2025/01/03 02:52:15 [debug] 548#548: *26 http copy filter: "/testapi?"

'''

How can I change the location and have nginx rewrite it correctly from the arbitrary name /testapi to the path and parameter that the service requires?

**Comment by Maryna-f5** [comment]
This discussion appears to focus on the configuration, troubleshooting or general questions around running NGINX, which is not the intended use for GitHub discussions. 

We encourage you to visit [https://community.nginx.org ](https://community.nginx.org/)and share your discussion there as the NGINX Community Forum is the most appropriate place to engage for these types of questions.
